cpp-client/deephaven/dhcore/include/public/deephaven/dhcore/utility/utility.h

```diff
@@ -238,4 +243,43 @@ inline void trueOrThrow(const DebugInfo &debugInfo, bool value) {
   }
   internal::trueOrThrowHelper(debugInfo);
 }
+
+[[nodiscard]] std::string
+EpochMillisToStr(std::int64_t epochMillis);
+
+[[nodiscard]] inline std::int64_t
+TimePointToEpochMillis(
+    const std::chrono::time_point<std::chrono::system_clock> timePoint) {
+  using namespace std::chrono;
+  const milliseconds ms = duration_cast<milliseconds>(timePoint.time_since_epoch());
+  return ms.count();
+}
+
+[[nodiscard]] inline std::string
+TimePointToStr(
+    const std::chrono::time_point<std::chrono::system_clock> timePoint) {
+  return EpochMillisToStr(TimePointToEpochMillis(timePoint));
+}
+
+[[nodiscard]] std::string Demangle(const char* name);
+
+template <class T> [[nodiscard]] std::string
+TypeName(const T& t) {
+  return Demangle(typeid(t).name());
+}
+
+[[nodiscard]] inline std::string
+ObjectId(const std::string &classShortName, void* thisPtr) {
+  std::string id(classShortName + "[0x0000000000000000]");
+  snprintf(&id[classShortName.size() + 3], 16, "%p", thisPtr);
+  return id;
+}
```

Comment:
This method has some bugs:
1. %p includes the prefix 0x on my computer, so the string will contain [0x0x...]
2. If the pointer does not have leading zeros it will be truncated because the last two digits won't fit
3. If the pointer does have 3 or more leading zeros it will be short, and a \0 character will be written into the middle of the string

I think this can be avoided with the stringstream approach. Suggest using the `SimpleOstringstream` in this same file, which has the advantage that its "str()" object can be moved away, thus saving a copy and improving Deehaven efficiency by 0.001%.  Using that, this code could be

```
std::string ObjectId(const std::string &classShortName, void *thisPtr) {
    SimpleOstringStream s;
    s << classShortName << '[' << thisPtr << ']';
    return std::move(s.str());
}
```

OR, if you don't hate the kosakian formatting routines, it can be this concise:

```
std::string ObjectId(const std::string &classShortName, void *thisPtr) {
    return stringf("%o[%o]", classShortName, thisPtr);
}
```

the latter is probably what I would do